compass/landice/tests/mismipplus/setup_mesh.py

```diff
@@ -0,0 +1,301 @@
+import warnings
+
+import numpy as np
+import xarray as xr
+from mpas_tools.io import write_netcdf
+from mpas_tools.logging import check_call
+from mpas_tools.mesh.conversion import convert, cull
+from mpas_tools.planar_hex import make_planar_hex_mesh
+
+from compass.model import make_graph_file
+from compass.step import Step
+
+
+class SetupMesh(Step):
+    """
+    A step for creating a mesh and initial condition for MISMIP+ test cases
+
+    Attributes
+    ----------
+    resolution : int
+        The resolution of the test case, as defined configuration file at the
+        when `compass setup` is run.
+
+    nx : int
+        The number of cells in the x direction
+
+    ny : int
+        The number of cells in the y direction
+
+    dc : int
+        The distance in meters between adjacent cell centers.
+
+    """
+    def __init__(self, test_case, resolution):
+        """
+        Create the step
+
+        Parameters
+        ----------
+        testcase : compass.TestCase
+            The test case this step belongs to
+
+        resolution : int
+            The distance between horizontal grid points
+        """
+        # Defined resolutions, the value of `resolution` in the configuration
+        # file needs to a key in the dictionary, or added manually.
+        resolution_params = {'8km': {'nx': 73, 'ny': 14, 'dc': 9237.604307},
+                             '4km': {'nx': 144, 'ny': 24, 'dc': 4618.802154},
+                             '2km': {'nx': 284, 'ny': 44, 'dc': 2309.401077},
+                             '1km': {'nx': 566, 'ny': 84, 'dc': 1154.700538}}
+
+        # With the defined resolution, make the associated key for the param
+        # dictionary (neccessary b/c unstructured hex meshes)
+        key = f'{resolution}km'
+
+        # Ensure the resolution passed in the configuration file is defined
+        # within the parameter dictionary. If this is not the case, either:
+        #   1. choose an already defined resolution
+        #   2. add the appropriate parameter values to the dict.
+        assert key in resolution_params
+
+        super().__init__(test_case=test_case,
+                         name=f'{key}_mesh_gen',
+                         subdir=f"{key}/mesh_gen")
+
+        # Files to be created as part of the this step
+        for filename in ['mpas_grid.nc', 'graph.info', 'landice_grid.nc']:
+            self.add_output_file(filename=filename)
+
+        # unpack the mesh parameters for the given resolution as set as
+        # attributes
+        self.nx, self.ny, self.dc = resolution_params[key].values()
```

Comment:
Good catch, I didn't necessarily think of this. But it turns out that since python 3.6+ dictionaries are now ordered in "insertion order". [Here's a description](https://docs.python.org/3.6/whatsnew/3.6.html#new-dict-implementation) of the changes that were implemented: 

> The [dict](https://docs.python.org/3.6/library/stdtypes.html#typesmapping) type now uses a “compact” representation based on [a proposal by Raymond Hettinger](https://mail.python.org/pipermail/python-dev/2012-December/123028.html) which was [first implemented by PyPy](https://morepypy.blogspot.com/2015/01/faster-more-memory-efficient-and-more.html). The memory usage of the new [dict()](https://docs.python.org/3.6/library/stdtypes.html#dict) is between 20% and 25% smaller compared to Python 3.5.
> 
> The order-preserving aspect of this new implementation is considered an implementation detail and should not be relied upon (this may change in the future, but it is desired to have this new dict implementation in the language for a few releases before changing the language spec to mandate order-preserving semantics for all current and future Python implementations; this also helps preserve backwards-compatibility with older versions of the language where random iteration order is still in effect, e.g. Python 3.5).

But given that it says the order preserving aspect shouldn't be relied upon, I'll switch to an `OrderedDict` so that I can keep the concise implementation.

Reply:
### Option 1
Well maybe`OrderDict` isn't the easiest way. While it works, it makes it a little harder to read. Here's what would be needed: 
```python
resolution_params = {'8km': OrderedDict([('nx', 73),  ('ny', 14), ('dc', 9237.604307)]),
                     '4km': OrderedDict([('nx', 144), ('ny', 24), ('dc', 4618.802154)]),
                     '2km': OrderedDict([('nx', 284), ('ny', 44), ('dc', 2309.401077)]),
                     '1km': OrderedDict([('nx', 566), ('ny', 84), ('dc', 1154.700538)])}
```
then 
```python
self.nx, self.ny, self.dc = resolution_params[key].values()
```
should work version independent. 

### Option 2
An alternative and easier method as compared to option 1 is: 
```python
mesh_params_dict = resolution_params[key].values()

for resolution_param in mesh_params_dict: 
    setattr(self, resolution_param, mesh_params_dict[mesh_params_dict])
``` 
but still not super easy to read. 

### Option 3
There's always the brute force option, which while not super concise is very legible:
```
mesh_params_dict = resolution_params[key].values()

self.dx = mesh_params_dict['dx']
self.dy = mesh_params_dict['dy']
self.dc = mesh_params_dict['dc']
```
---
While this isn't super important for the PR, which do you prefer?